test/terra/backends/aer_simulator/test_fusion.py

```diff
@@ -511,3 +512,45 @@ def test_fusion_diagonal(self):
             if op_name == 'measure':
                 break
             self.assertEqual(op_name, 'diagonal')
+
+
+    def test_parallel_fusion_diagonal(self):
+        """Test diagonal fusion with parallelization"""
+        backend = self.backend(method="statevector")
+        
+        num_of_qubits = 10
+        circuit = QuantumCircuit(num_of_qubits)
+        size = 3
+        
+        for q in range(num_of_qubits):
+            circuit.h(q)
+        
+        for qubit in range(num_of_qubits):
+            for q in range(size):
+                ctrl = (qubit + q) % num_of_qubits
+                tgt = (ctrl + 1) % num_of_qubits
+                circuit.cx(ctrl, tgt)
+        
+            circuit.p(np.random.random(), tgt)
+        
+            for q in range(size):
+                ctrl = (qubit + size - q - 1) % num_of_qubits
+                tgt = (ctrl + 1) % num_of_qubits
+                circuit.cx(ctrl, tgt)
+                
+        circuit.save_statevector()
+        
+        thread = 8
+        result = backend.run(circuit,
+                               fusion_threshold=1,
+                               **{'fusion_enable.diagonal': True,
+                                  'fusion_enable.cost_based': False,
+                                  'fusion_enable.n_qubits': False},
```

Comment:
I had no idea you could pass invalid identifier names like this!  That's cool, although I'm not certain it's the best user interface...  Still, that's beyond the scope of this PR.

Reply:
'fusion_enable.*' options are just for debug use. If users require to control fusion strategies, we should rename option names :-)